client-tests/src/servers/testServer.ts

```diff
@@ -214,5 +221,18 @@ connection.onTypeDefinition((params) => {
 	return { uri: params.textDocument.uri, range: { start: { line: 2, character: 2}, end: {line: 3, character: 3 }}};
 });
 
+connection.onRequest(regularCancellationTestType, async (token) => {
+	while (!token.isCancellationRequested) {
+		await delay(0);
```

Comment:
@dbaeumer regular cancellation doesn't work without async since it requires code to go back to the event loop to handle cancellation requests.

the point of file-based cancellation is that it works with a synchronous code path and the test is confirming that is working as expected. 

if you remove "await" in the regular cancellation test, the test hang. on the contrary, file-based cancellation works as expected.

Reply:
Good point. Please add a comment so that we know why this is the case.